### PR TITLE
feat: added multi-transaction context ux to marketplace actions

### LINKED ISSUE #337: UX for transactions
As part of the work on profile stores and nft details buy/sell a good flow has been established for approving transactions and giving feedback to user, a few transactions still exist in the buy/sell flow that don't utilize this context
- [x] Cancel listing
- [x] Cancel offer
- [x] Make offer
- [x] Buy now
- [x] Make listing

### PR DESCRIPTION
**Pull request by milorue**
closes #337

### CODE REVIEW

File: src/common/components/forms/CancelOfferForm.tsx [hunk 1]
```diff
@@ -48,10 +52,31 @@ const CancelOfferForm: FC<CancelOfferFormProps> = ({
...
+      },
+    ];
+
     try {
-      await onCancelOffer();
-      toast.success(`Confirmed cancel offer success`);
-      await refetch();
+      await runActions(newActions, {
+        onActionSuccess: async () => {
+          toast.success(`Confirmed cancel offer success`);
```
Comment (austinmilt) [inline src/common/components/forms/CancelOfferForm.tsx:68]: Is the order important? In other components you show success _after_ `await refetch();`

File: src/common/components/forms/CancelOfferForm.tsx [hunk 2]
```diff
@@ -48,10 +52,31 @@ const CancelOfferForm: FC<CancelOfferFormProps> = ({
...
+          await refetch();
+        },
+        onActionFailure: async (err) => {
+          toast.error(err.message);
```
Comment (austinmilt) [inline src/common/components/forms/CancelOfferForm.tsx:72]: In other components you dont display the error inside `onActionFailure`, only in the outer `catch`. Is that intentional?

File: src/common/components/forms/CancelSellForm.tsx [hunk 3]
```diff
@@ -54,12 +57,33 @@ const CancelSellForm: FC<CancelSellFormProps> = ({
...
+          await refetch();
+          toast.success(`Confirmed cancel listing success`);
+        },
+        onComplete: async () => {
+          await refetch();
+          setOpen(false);
+        },
+        onActionFailure: async (err) => {
+          await refetch();
+          // toast.error(err.message);
```
Comment (austinmilt) [inline src/common/components/forms/CancelSellForm.tsx:82]: remove commented code?

File: src/common/components/forms/CancelSellForm.tsx [hunk 4]
```diff
@@ -54,12 +57,33 @@ const CancelSellForm: FC<CancelSellFormProps> = ({
...
+        },
+      });
     } catch (err: any) {
-      toast.error(err.message);
+      // toast.error(err.message);
```
Comment (austinmilt) [inline src/common/components/forms/CancelSellForm.tsx:86]: remove commented code?

File: src/common/components/forms/OfferForm.tsx [hunk 5]
```diff
@@ -75,13 +78,29 @@ const OfferForm: FC<OfferFormProps> = ({ nft, marketplace, refetch }) => {
...
+        },
+        onComplete: async () => {
+          await refetch();
+        },
+        onActionFailure: async () => {
+          await refetch();
+        },
+      });
     } catch (err: any) {
-      toast.error(err.message);
```
Comment (austinmilt) [inline src/common/components/forms/OfferForm.tsx:84]: This appears to never display the error. Intentional?

**Review by austinmilt — APPROVED** [review]
some things to consider changing, but looks fine all-in-all

File: src/common/components/forms/CancelOfferForm.tsx (hunk 1 above)
Comment (milorue) [reply src/common/components/forms/CancelOfferForm.tsx:68]: order shouldn't matter

File: src/common/components/forms/CancelOfferForm.tsx (hunk 2 above)
Comment (milorue) [reply src/common/components/forms/CancelOfferForm.tsx:72]: the inner throws errors that are intended the external will catch unintended errors

File: src/common/components/forms/CancelSellForm.tsx (hunk 3 above)
Comment (milorue) [reply src/common/components/forms/CancelSellForm.tsx:82]: y

File: src/common/components/forms/OfferForm.tsx (hunk 5 above)
Comment (milorue) [reply src/common/components/forms/OfferForm.tsx:84]: an oversight thx for catching it

**Pull request merged**